python/pyspark/sql/types.py

```diff
@@ -1617,7 +1617,7 @@ def to_arrow_type(dt):
     elif type(dt) == DoubleType:
         arrow_type = pa.float64()
     elif type(dt) == DecimalType:
-        arrow_type = pa.decimal(dt.precision, dt.scale)
+        arrow_type = pa.decimal128(dt.precision, dt.scale)
```

Comment:
@wesm @BryanCutler Is this a right way to define decimal type for Arrow?
I also wonder if there is a limit for precision and scale?

Reply:
yes, that's the right way - it is now fixed at 128 bits internally.  I believe the Arrow Java limit is the same as Spark 38/38, not sure if pyarrow is the same but I think so.